tool/tsh/common/tsh.go

```diff
@@ -812,6 +818,9 @@ func Run(ctx context.Context, args []string, opts ...CliOption) error {
 	app.Flag("cert-format", "SSH certificate format").StringVar(&cf.CertificateFormat)
 	app.Flag("trace", "Capture and export distributed traces").Hidden().BoolVar(&cf.SampleTraces)
 	app.Flag("trace-exporter", "An OTLP exporter URL to send spans to. Note - only tsh spans will be included.").Hidden().StringVar(&cf.TraceExporter)
+	// This flag only applies to tsh ssh; it's defined here to make configuring
+	// the re-exec command easier.
+	app.Flag("fork-signal-fd", "File descriptor to signal parent on when forked. Overrides --fork-after-authentication. For internal use only.").Hidden().Uint64Var(&cf.forkSignalFd)
```

Comment:
Either we check for `forkSignalFd > 2` (rather than checking for nonzero) or we set it to a negative number by default and check that it's not negative. My vote is to require `> 2` but if you go for the latter:

```suggestion
	// This flag only applies to tsh ssh; it's defined here to make configuring
	// the re-exec command easier.
	app.Flag("fork-signal-fd", "File descriptor to signal parent on when forked. Overrides --fork-after-authentication. For internal use only.").Hidden().Default("-1").Uint64Var(&cf.forkSignalFd)
```

Reply:
I believe >2 won't work for Windows, since 1 and 2 would be acceptable file descriptors. I made the check OS-specific.